app/controllers/admin/pages_controller.rb

```diff
@@ -23,14 +23,25 @@ def edit
 
     def update
       @page = Page.find(params[:id])
-      @page.update!(page_params)
-      redirect_to "/admin/pages"
+      @page.assign_attributes(page_params)
+      if @page.valid?
+        @page.update!(page_params)
+        redirect_to admin_pages_path
+      else
+        flash.now[:error] = @page.errors.full_messages.join(", ")
```

Comment:
I like using `model.errors_as_sentence` :)

https://github.com/forem/forem/blob/531f26bad69e6cb03468d1f84d2e4a64c3dea757/app/models/application_record.rb#L56-L58

Reply:
Also curious, why `flash.now` over regular `flash`? So that it doesn't have a chance to persist?

Reply:
Cool, I'll do that! Thanks!

`flash` uses the flash message on the next request (so would be ideal before `redirect_to`), whereas `flash.now` will use the flash message on the current request (so generally, called before `render`).